merkletree/proof.go

```diff
@@ -16,6 +23,75 @@ type ProofNode interface {
 	Commitment() []byte
 }
 
+func computeLeafHash(ap *AuthenticationPath,
+	leafIndex, leafCommitment []byte, leafLevel int, isLeafEmpty bool) (leafHash []byte) {
+	if isLeafEmpty {
```

Comment:
> I wonder if it's worth going back and changing the STR and TB verify functions for this reason. Maybe if we notice the memory footprint of these functions being too big, we should.

In case of STR and TB verification, I think it is a trade-off here. We have either advantages in memory footprint or cpu computations, but not both. Am I right?